examples/split_plot_rcbd.R

```diff
@@ -0,0 +1,214 @@
+# This is an example of a split-plot RCBD experimental design based off of the first
+# example in [1] and further analysis in [2]. The data can be downloaded from
+# [3].
+#
+# To run the script type:
+#
+# source('split_plot_rcbd.R', print.eval = TRUE)
+#
+# [1] http://www.unh.edu/halelab/BIOL933/labs/lab9.pdf
+# [2] http://www.personal.psu.edu/mar36/stat_461/split_plot/split_plot.html
+# [3] http://www.unh.edu/halelab/BIOL933/schedule.htm
+#
+# TODO : Maybe I should use the `data('plots')` from the agricolae package
+# instead?
+#
+# NOTE : The code that will be shown to the user in the app is bounded by these
+# separators: #------------#.
+
+# Load the necessary libraries.
+#-----------------------------------------------------------------------------#
+library('agricolae')
+library('effects')  # for allEffects()
+#-----------------------------------------------------------------------------#
+
+sep <- function(n){
+  # This function simply prints a line of "=" to the screen as a separator and
+  # will not show up in the app. It is only here for nice console printing.
+  line <- paste0(paste(replicate(n, "="), collapse = ""), '\n')
+  cat(line)
+}
+
+# Load the data.
+# repetitions: block
+# main plots: SeedLotA (four plots in a block)
+# subplots: TrtmtB (four subplots in a main plot)
+#-----------------------------------------------------------------------------#
+my.data <- read.csv('Lab9ex1.csv')
+#-----------------------------------------------------------------------------#
+
+# The factors in this experiment are recorded with numerical values so they need
+# to be recoded as factors. The user will have to set this manually in the app.
+# NOTE : This is not necessary if `aov()` is used instead of `lm()`.
+#-----------------------------------------------------------------------------#
+my.data$SeedLotA <- as.factor(my.data$SeedLotA)
+my.data$Block <- as.factor(my.data$Block)
+my.data$TrtmtB <- as.factor(my.data$TrtmtB)
+#-----------------------------------------------------------------------------#
+
+##Testing various transformations for improving assumption tests##
+#(1) Create a sqrt-transformed variable
+my.data[,4]<-sqrt(my.data[,4])
+
+#(2) Create a log-transformed variable
+my.data[,4]<-log10(my.data[,4])
+
+#(3)----- Finding the exponent for a power transformation ---- #
+my.data$merged_treatment <- paste(my.data$SeedLotA, my.data$TrtmtB, sep = "-")
+as.factor(my.data$merged_treatment)
+str(my.data)
+means <- aggregate(my.data$Yield, list(my.data$merged_treatment), mean)
+vars <- aggregate(my.data$Yield, list(my.data$merged_treatment), var)
+logmeans <- log10(means$x)
+logvars <- log10(vars$x)
+power.mod<-lm(logvars ~ logmeans)
+summary<-summary(power.mod)
+#identify the slope
+summary$coefficients[2,1]
+#calculate the appropriate power of the transformation, where Power = 1 – (slope/2)
+power <- 1-(summary$coefficients[2,1])/2
+power
+#Create power-tranformed variable
+my.data$Yield<-(my.data$Yield)^(power)
+
```

Comment:
@msimmond  These lines effectively do a combined transformation, which is equivalent to:

`my.data$Yield <- log10(sqrt(my.data$Yield))^power`

I'm guessing that is not what you intended. Does this example need a transformation, and if so which one?